lib/index.js

```diff
@@ -191,24 +191,31 @@ internals.root = function () {
     root.reach = function (schema, path) {
 
         Hoek.assert(schema && schema instanceof Any, 'you must provide a joi schema');
-        Hoek.assert(Array.isArray(path), 'path must be an array');
+        Hoek.assert(Array.isArray(path) || typeof path === 'string', 'path must be a string or an array of strings');
 
-        if (!path.length) {
-            return schema;
-        }
+        const reach = (sourceSchema, schemaPath) => {
 
-        const children = schema._inner.children;
-        if (!children) {
-            return;
-        }
+            if (!path.length) {
+                return sourceSchema;
+            }
 
-        const key = path.shift();
-        for (let i = 0; i < children.length; ++i) {
-            const child = children[i];
-            if (child.key === key) {
-                return this.reach(child.schema, path);
+            const children = sourceSchema._inner.children;
+            if (!children) {
+                return;
             }
-        }
+
+            const key = schemaPath.shift();
+            for (let i = 0; i < children.length; ++i) {
+                const child = children[i];
+                if (child.key === key) {
+                    return this.reach(child.schema, schemaPath);
+                }
+            }
+        };
+
+        const schemaPath = typeof path === 'string' ? path.split('.') : path;
```

Comment:
Can you call `.slice()` in the 2nd case ? I don't want to modify input parameters.